lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt

```diff
@@ -99,14 +96,44 @@ class GatewayProvider(
         return response.body
     }
 
-    private fun <T> buildDeserializer(deserializationStrategy: DeserializationStrategy<T>): HttpResponseDeserializer<T> =
+    private fun <T> buildDeserializer(
+        deserializationStrategy: DeserializationStrategy<T>,
+    ): HttpResponseDeserializer<T> =
         Function { response ->
             val body = handleResponseError(response)
 
-            val json = Json { ignoreUnknownKeys = true }
             json.decodeFromString(deserializationStrategy, body)
         }
 
+    private fun handleMissingTransaction(response: HttpResponse): String {
+        val body = handleResponseError(response)
+
+        @Serializable
+        data class MissingTransactionResponse(val status: TransactionStatus)
+
+        try {
+            val missingTransaction = Json.decodeFromString(MissingTransactionResponse.serializer(), body)
+            when (missingTransaction.status) {
+                TransactionStatus.UNKNOWN, TransactionStatus.PENDING -> throw GatewayRequestFailedException(
+                    message = "Transaction not received or unknown",
+                    payload = body,
+                )
+
+                else -> {}
+            }
+        } catch (_: SerializationException) {
+            // Transaction wasn't exactly a MissingTransactionResponse, continue parsing
+        }
```

Comment:
I guess it can be simplified. My original idea was to explicitly only handle transactions with format
```json
{
    "status": "NOT_RECEIVED"
}
``` 
and alike here, but it doesn't change much I guess.